spot-client/src/common/remote-control/xmpp-connection.js

```diff
@@ -653,25 +682,33 @@ export default class XmppConnection {
         const from = presence.getAttribute('from');
         let type = presence.getAttribute('type');
 
+        // Assume unknown participant presence indicates a join.
         if (!this._participants.has(from) && !type) {
             type = 'join';
         }
 
-        return {
-            from,
-            localUpdate: from === this.getRoomFullJid(),
-            state: Array.from(presence.children)
-                .reduce((acc, child) => {
-                    let value = child.textContent;
+        const state = Array.from(presence.children)
```

Comment:
Here will be a merge config from https://github.com/jitsi/jitsi-meet-spot/pulls?q=is%3Apr+is%3Aclosed+author%3Apaweldomas

Reply:
Fixed it but looks like I'll have to send the kicked reasons in some other way, because it's being received as xml.

Reply:
Instead of changing the way it's sent I changed how it's parsed, using the info you told me about.